assets/src/components/mapPage.tsx

```diff
@@ -57,70 +65,298 @@ const ToggleMobileDisplayButton = ({
   )
 }
 
-const MapPage = (): ReactElement<HTMLDivElement> => {
-  const [{ searchPageState, mobileMenuIsOpen }, dispatch] =
-    useContext(StateDispatchContext)
-  const stations = useStations()
-
+const SearchInputAndResults = ({
+  searchPageState,
+  mobileDisplay,
+  selectedVehicleId,
+  selectVehicle,
+}: {
+  searchPageState: SearchPageState
+  mobileDisplay?: ReactElement
+  selectedVehicleId: string | null
+  selectVehicle: (vehicle: VehicleOrGhost | null) => void
+}): React.ReactElement => {
   const { socket }: { socket: Socket | undefined } = useContext(SocketContext)
-  const vehicles: VehicleOrGhost[] | null = useSearchResults(
+  const searchVehicles: VehicleOrGhost[] | null = useSearchResults(
     socket,
     searchPageState.isActive ? searchPageState.query : null
   )
-  const onlyVehicles: Vehicle[] = filterVehicles(vehicles)
-  const [mobileDisplay, setMobileDisplay] = useState(MobileDisplay.List)
-  const [selectedVehicleId, setSelectedVehicleId] = useState<VehicleId | null>(
-    searchPageState.selectedVehicleId ?? null
+
+  return (
+    <>
+      <div className="m-map-page__input">
+        <SearchForm formTitle="Search Map" inputTitle="Search Map Query" />
+        {mobileDisplay}
+      </div>
+
+      <hr />
+
+      <div className="m-search-display">
+        {searchVehicles !== null &&
+        thereIsAnActiveSearch(searchVehicles, searchPageState) ? (
+          <SearchResults
+            vehicles={searchVehicles}
+            selectedVehicleId={selectedVehicleId}
+            onClick={selectVehicle}
+          />
+        ) : (
+          <RecentSearches />
+        )}
+      </div>
+    </>
   )
-  const [searchOpen, setSearchOpen] = useState<boolean>(true)
+}
 
-  const liveVehicle: Vehicle | null = selectedVehicleId
-    ? onlyVehicles.find((v) => v.id === selectedVehicleId) || null
-    : null
-  const [showVehicleCard, setShowVehicleCard] = useState<boolean>(
-    searchPageState.selectedVehicleId ? true : false
+// #region Map Display
+const useCachedVehicleById = (
```

Comment:
when `useVehicleForId` changes which topic it's subscribed to, it initially returns `null` which makes the VPC and route shape, route vehicles, and selected vehicle, blip in and out of existance as it rapidly rerenders.

I cache the last successful result before `selectedVehicleId` changes so that we don't blip the UI while loading the next vehicle.
I had this working with a reducer/state, but because what triggers the render is `useVehicleForId` updating, we only need to hold onto the previous vehicle till the next one is available/not null WHILE the `selectedVehicleId` value tells us that we didn't deselect a vehicle by not being `null`.

Maybe naming this `useDiferredVehicleById` would be better?... though this doesn't match what `useDiferredValue` does exactly. (I also tried to get this to work with `useDiferredValue` but that didn't work because `useSelectedVehicleForId` may return `null` multiple times before the next value/vehicle has resolved)

Reply:
Just spitballing here, how about `useMostRecentVehicleById`?

Reply:
@lemald 👍🏻👍🏻

Reply:
@mbta/skate-developers please take a look at [d90c0e53](https://github.com/mbta/skate/pull/1891/commits/d90c0e53016f508ce9a39a7d24ebc82fff14a66a) 🙏🏻

Reply:
This looks good to me! I'm still working on understanding the benefits of `useRef` over `useState` in this case, but maybe I just need to revisit the `useRef` docs & don't want to block merging over it!

Reply:
One nitpick-y comment about variable names but otherwise I like it!

Reply:
> benefits of useRef over useState in this case

Here's my understanding on what would happen if we used `useState` instead of `useRef`. 

When the `useVehicleForId` hook updates asynchronously, the state update causes `<MapDisplay/>` to rerender.

On that next render, `MapDisplay` can save the result of `useVehicleForId` using either `useState` or `useRef`.

When `useVehicleForId` updates, we need to store the value in case the next render returns `null` (which should only happen when `selectedVehicleId` changes and is not `null`). [^1]


__If `useVehicleForId` is cached in a `useState`, then updating the cache will cause a rerender, which would mean for every `useVehicleForId` update we would render the `<MapDisplay/>` twice.__



[^1]: Actually, maybe we can catch the last return value of `useVehicleForId` when `selectedVehicleId` changes? But unless the lambda in a `useEffect(..., [selectedVehicleId])` can capture the `VehicleOrGhost` before/while it changes to `null`, I still think we'd need a ref or state to track `lastSelectedVeihcleId`?

Reply:
I'm still learning react so maybe there's something I missed! but 'TLDR;' I picked `useRef` because I didn't want to cause a rerender when updating it.